File: code/android/app/src/main/java/net/mstoegerer/tasknest/dto/TodoDto.kt
package net.mstoegerer.tasknest.dto

import java.util.*

data class TodoDto(
    val id: UUID,
    val title: String,
    val content: String,
    val createdUtc: Date,
    val updatedUtc: Date,
    val status: String,
    val userId: UUID,
    val assignedToId: UUID,
    val location: PointDto?,
    val attachments: List<AttachmentDto> = emptyList()
)

File: code/backend/net.mstoegerer.TaskNest/net.mstoegerer.TaskNest.Api/Domain/DTOs/TodoDto.cs
namespace net.mstoegerer.TaskNest.Api.Domain.DTOs;

public class TodoDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public string Content { get; set; } = null!;
    public DateTime CreatedUtc { get; set; }
    public DateTime UpdatedUtc { get; set; }
    public string Status { get; set; } = null!;
    public Guid UserId { get; set; }
    public DateTime? DueUtc { get; set; }
    public Guid AssignedToId { get; set; }
    public PointDto? Location { get; set; }
    public ICollection<AttachmentDto> Attachments { get; set; } = new HashSet<AttachmentDto>();
}